projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js

```diff
@@ -0,0 +1,433 @@
+
+	const fileName = track.tmpFile ? track.tmpFile.name : '';
+
+	const mediaUpload = useSelect( select => {
+		return select( blockEditorStore ).getSettings().mediaUpload;
+	}, [] );
+
+	if ( ! mediaUpload ) {
+		return null;
+	}
+
+	return (
+		<NavigableMenu>
+			<div className="videopress-block-tracks-editor__single-track-editor">
+				<span className="videopress-block-tracks-editor__single-track-editor-label">
+					{ __( 'Edit track', 'jetpack' ) }
+				</span>
+				<div className="videopress-block-tracks-editor__single-track-editor-edit-file-label">
+					<span>{ __( 'File', 'jetpack' ) }:</span>
+					{ '' !== fileName && (
+						<span className="videopress-block-tracks-editor__single-track-editor-edit-file-label-name">
+							<strong>{ fileName }</strong>
+						</span>
+					) }
+					<MediaUploadCheck>
+						<FormFileUpload
+							onChange={ event => {
+								const files = event.target.files;
+
+								if ( ! files.length > 0 ) {
+									return;
+								}
+
+								track.tmpFile = files[ 0 ];
+								onChange( track );
+							} }
+							accept=".vtt,text/vtt"
+							render={ ( { openFileDialog } ) => {
+								return (
+									<Button
+										isLink
+										onClick={ () => {
+											openFileDialog();
+										} }
+									>
+										{ '' === fileName
+											? __( 'Select track', 'jetpack' )
+											: __( 'Change track', 'jetpack' ) }
+									</Button>
+								);
+							} }
+							disabled={ isSavingTrack }
+						/>
+					</MediaUploadCheck>
+				</div>
+				<div className="videopress-block-tracks-editor__single-track-editor-label-language">
+					<TextControl
+						/* eslint-disable jsx-a11y/no-autofocus */
+						autoFocus
+						/* eslint-enable jsx-a11y/no-autofocus */
+						onChange={ newLabel =>
+							onChange( {
+								...track,
+								label: newLabel,
+							} )
+						}
+						label={ __( 'Label', 'jetpack' ) }
+						value={ label }
+						help={ __( 'Title of track', 'jetpack' ) }
+						disabled={ isSavingTrack }
+					/>
+					<TextControl
+						onChange={ newSrcLang =>
+							onChange( {
+								...track,
+								srcLang: newSrcLang,
+							} )
+						}
+						label={ __( 'Source language', 'jetpack' ) }
+						value={ srcLang }
+						help={ __( 'Language tag (en, fr, etc.)', 'jetpack' ) }
+						disabled={ isSavingTrack }
+					/>
+				</div>
+				<SelectControl
+					className="videopress-block-tracks-editor__single-track-editor-kind-select"
+					options={ KIND_OPTIONS }
+					value={ kind }
+					label={ __( 'Kind', 'jetpack' ) }
+					onChange={ newKind => {
+						onChange( {
+							...track,
+							kind: newKind,
+						} );
+					} }
+					disabled={ isSavingTrack }
+				/>
+				<div className="videopress-block-tracks-editor__single-track-editor-buttons-container">
+					{ isSavingTrack ? (
+						<Spinner />
+					) : (
+						<Button
+							isSecondary
+							disabled={ ! track.tmpFile }
+							onClick={ () => {
```

Comment:
Maybe this could be extracted? I'm thinking it would be easier to read. What do you think?